src/libraries/System.Private.Xml/src/System/Xml/Schema/DataTypeImplementation.cs

```diff
@@ -1088,6 +1089,8 @@ internal override RestrictionFlags ValidRestrictionFlags
             return exception;
         }
 
+        [UnconditionalSuppressMessage("AotAnalysis", "IL3050:AotUnfriendlyApi",
+            Justification = "All types that are instantiated with this method are used elsewhere in this file in the implementations of the DatatypeImplementation abstract class.")]
```

Comment:
I assume that this warning is about `values.ToArray(_itemType.ValueType)`. I think it works, but it fragile and hard to see that it is correct.

I would suggest factoring out the code that produces the Aot warning into a local method, apply the warning onto the local method only, and use the array type to create the array to guarantee that it is kept. Something like:

```csharp
array = CreateArray(ListValueType);
values.CopyTo(array, 0);

...
[UnconditionalSuppressMessage("AotAnalysis", "Array type is passed as an argument to this method, so it is guaranteed to exist.")]
static Array CreateArray(Type arrayType, int length)
{
    Array array = Array.CreateInstance(arrayType.GetElementType()!, length);
    Debug.Assert(array.GetType() == arrayType);
    return array;
}
```

Reply:
Opened https://github.com/dotnet/runtime/issues/76478 to make this better.